src/main/java/org/broadinstitute/hellbender/tools/exome/IntervalBackedExonCollection.java

```diff
@@ -1,30 +1,60 @@
 package org.broadinstitute.hellbender.tools.exome;
 
-import org.broadinstitute.hellbender.utils.GenomeLoc;
 import org.broadinstitute.hellbender.utils.IndexRange;
+import org.broadinstitute.hellbender.utils.SimpleInterval;
 
 import java.util.*;
+import java.util.stream.Collectors;
+import java.util.stream.IntStream;
 
 /**
  * Exon data-base based on a list of intervals.
  *
  * @author Valentin Ruano-Rubio &lt;valentin@broadinstitute.org&gt;
  */
-public final class IntervalBackedExonCollection implements ExonCollection<GenomeLoc> {
+public final class IntervalBackedExonCollection implements ExonCollection<SimpleInterval> {
 
     /**
      * Map from interval name to interval object.
      */
-    private final Map<String,GenomeLoc> intervalsByName;
+    private final Map<String,SimpleInterval> intervalsByName;
 
     /**
      * Sorted list of intervals.
      */
-    private final List<GenomeLoc> sortedIntervals;
+    private final List<SimpleInterval> sortedIntervals;
+
+    /**
+     * Compares intervals returning 0 only if they are exactly the same.
+     *
+     * <p>
+     *    Given intervals {@code a} and {@code b} on the same contig, this comparator will return:
+     *     <ul>
+     *         <li>-1 if {@code a}'s start is before {@code b}'s and if the same, {@code a}'s end is before {@code b}'s,</li>
+     *         <li>1 if {@code b}'s start is before {@code a}'s and if the same, {@code b}'s end is before {@code a}'s or</li>
```

Comment:
This description is really confusing.   It sounds like the algorithm is
```
if (a.start < b.start && a.end < b.end) )
    return -1
else if (b.start < a.start && b.end < a.end)
    return 1
else
   return 0
```

Which is not the what the comparator is doing.  

It would be clearer to say something like "First compare the start position of both intervals, if they're different compare then end position" or just literally copy the code from the actual comparator.